packages/api/src/beacon/server/events.ts

```diff
@@ -49,7 +50,12 @@ export function getRoutes(config: ChainForkConfig, api: ServerApi<Api>): ServerR
             // The client may disconnect and we need to clean the subscriptions.
             req.raw.once("close", () => resolve());
             req.raw.once("end", () => resolve());
-            req.raw.once("error", (err) => reject(err));
+            req.raw.once("error", (err) => {
+              if ("code" in err && (err as unknown as {code: string}).code === "ECONNRESET") {
```

Comment:
I am not really sure if this is ideal, we might swallow cases where validator client crashed or disconnected for other reasons than shutting it down.

Was really hoping that this could be somehow addressed on the validator client side, it should not cause `ECONNRESET` error on the beacon node but disconnect gracefully.

Reply:
We are not swallowing something, trying to log more relevant information. If the client crashed internally it's still aborted from the context of AbortController on client. So we can't differentiate on the server side of both scenarios.

Reply:
I haven't looked at the code specifically but shouldn't the validator client be able to more gracefully close the connection is it receives a abort signal?